py3status/modules/yubikey.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""
+Show an indicator when YubiKey is waiting for a touch.
+
+Configuration parameters:
+    cache_timeout: How often to detect a pending touch request.
+        (default 1)
+    format: Display format for the module.
+        (default '\?if=is_waiting&color=bad Y')
+    u2f_keys_path: Full path to u2f_keys if you want to monitor sudo access.
+        (default '~/.config/Yubico/u2f_keys')
+
+SAMPLE OUTPUT
+{'color': '#FF0000', 'full_text': 'Y'}
+
+Dependencies:
+    gpg: to check for pending gpg access request
+    inotify-simple: to check for pending sudo request
+    subprocess32: if after all these years you are still using python2
+
+@author Maxim Baz (https://github.com/maximbaz)
+@license BSD
+"""
+
+import time
+import os
+import signal
+import sys
+import threading
+import inotify_simple
+
+if os.name == 'posix' and sys.version_info[0] < 3:
+    import subprocess32 as subprocess
+else:
+    import subprocess
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 1
+    format = '\?if=is_waiting&color=bad Y'
+    u2f_keys_path = '~/.config/Yubico/u2f_keys'
+
+    def post_config_hook(self):
+        self.u2f_keys_path = os.path.expanduser(self.u2f_keys_path)
+
+        self.killed = threading.Event()
+
+        self.pending_gpg = False
+        self.pending_sudo = False
+        self.sudo_reset_timer = None
+
+        class GpgThread(threading.Thread):
```

Comment:
I don't know anything about threading, but I ponder if we need classes here... And if this could be done by using methods alone. @tobes can give you better feedbacks.

Reply:
I'm more concerned by the approaches that this code takes, these are very hacky tricks... I hope someone from Yubico would give a hint on how to make it better. Especially the check for `gpg`...